cmake/CompilerFlags.cmake

```diff
@@ -23,10 +18,16 @@ set(GCC_DEBUG
 set(GCC_RELEASE -O3)  # Level 3 optimisation. Could also consider -fpack-derived
 
 # Intel 
+set(INTEL_BASE
+    -stand f08          # Fortran standard set to 2008
+    -implicitnone       # Give warning when undeclared variable is used
+    -diag-disable=5268  # Remove warning: Extension to standard: The text exceeds right hand column allowed on the line.
+    -free               # Specify free-fortmat
+   )
```

Comment:
@minyez The general question is, which verison/s of ifort are these valid for? Ifort classic/LLVM looks like the flags are somewhat inconsistent with what's here.

@Panadestein  You fancy adding an Intel build to the CI matrix in this PR? Else nothing can be validated.

Reply:
These options are valid for both ifort and ifx (LLVM), tested in oneAPI versions 2022 update 1 and 2023 update 1. On the other hand, some check options specified in the release flags are not implemented in the early version of ifx.